lib/publiccloud/basetest.pm

```diff
@@ -51,16 +53,42 @@ sub provider_factory {
     else {
         die('Unknown PUBLIC_CLOUD_PROVIDER given');
     }
+
+    $provider->init();
+    $self->{provider} //= [];
+    push(@{$self->{provider}}, $provider);
+    return $provider;
 }
 
-sub get_image_id {
-    my ($self, $provider) = @_;
-    my $img_url    = get_required_var('PUBLIC_CLOUD_IMAGE_LOCATION');
-    my ($img_name) = $img_url =~ /([^\/]+)$/;
-    my $image_id   = $provider->find_img($img_name);
-    die("Image $img_name is not available in the cloud provider") unless ($image_id);
-    set_var('PUBLIC_CLOUD_IMAGE_ID', $image_id);
-    return $image_id;
+sub cleanup {
+    # to be overridden by tests
```

Comment:
die if not implemented/overridden??

Reply:
If the test didn't need to cleanup something special, then it should not override the method.